mozilla/blurts-server
# MNTOR-2022: Integrate Glean.js & hook up generic events

# References: 
Jira: MNTOR-2022
Figma: 




# Description
Adds and configures Glean.js for Monitor. Since we are using Glean.js in the front-end, there are some specific ways in which we need to use Glean.

There are a few sharp edges I tried to address
- Next.js has no entry file that runs for every app load, so Glean needs to be initialized for every route. To manage this, I created a custom `useGlean` hook that handles initialization and returns the objects you need to record data.
- You cannot record Glean data directly on a non-client component, making things like recording a page view difficult. A pattern that we can use to handle this would be something like the `PageLoad` component which you can drop onto any page with whatever props you would like to pass as event extras. Then on first mount, the `PageLoad` component will record an event and then not run again as long as it doesn't re-render. So putting this at the top-level of a component and making sure it isn't a child component of a re-render heavy component is important.
- Rather than set up individual pings, we can send events as they happen using Glean's `maxEvent: 1` capability. This will record events and send a ping with those events immediately, similar to recording an event in GA.


# Screenshot (if applicable)

Not applicable.

# How to test
1. `npm install`
2. `npm run build-glean`
3. `npm run dev`
4. Open the browser console
5. Observe a block that shows data about the ping that was just submitted (when loading the home page).
6. Open https://debug-ping-preview.firebaseapp.com
7. Find the debug tag, currently checked in tag is "monitor-testing"
8. Confirm that the ping UUID in the browser console can be found in the debug ping viewer (sometimes a ping can take a few minutes to show up).


# Checklist (Definition of Done)
